### ISSUE TITLE
MoLang: Test whether custom function is side effect free & then completely resolve it at compile time

### ISSUE THREAD

**Issue by solvedDev** [bridge-core/editor#153]
side effect free: Static arguments, no `var.` access, ...

**Comment by Tschipp** [comment]
Exactly. In my project, we have a lot of math based animations, that are in essence just some sin/cos with some constant factors, but those factors are often spread out over several additions and multiplications. This is currently already a feature, but it only optimizes from left to right, and not right to left. Doing both could save some operations.
